src/hooks/useTokenDisplayData.js

```diff
@@ -0,0 +1,60 @@
+// @flow
+
+import { type EdgeCurrencyConfig, type EdgeCurrencyWallet } from 'edge-core-js'
+import { type EdgePluginMap } from 'edge-core-js/types'
+
+import { useSelector } from '../types/reactRedux'
+import { getAllTokens } from '../util/CurrencyInfoHelpers'
+import { fixFiatCurrencyCode, getDenomFromIsoCode, getYesterdayDateRoundDownHour, zeroString } from '../util/utils'
+
+/**
+ * Returns data from tokens relevant for display
+ * */
+
+export const useTokenDisplayData = (props: { currencyConfigMap: EdgePluginMap<EdgeCurrencyConfig>, tokenId?: string, wallet: EdgeCurrencyWallet }) => {
+  const { currencyConfigMap, tokenId, wallet } = props
+  const { currencyInfo } = wallet
+  const nativeCurrencyCode = currencyInfo.currencyCode
+  const tokens = getAllTokens(currencyConfigMap[currencyInfo.pluginId])
+
+  // Get currencyCode and deomination from token
+  let currencyCode, denomination
+  if (tokenId == null || nativeCurrencyCode === tokens[tokenId].currencyCode) {
+    currencyCode = nativeCurrencyCode
+    denomination = currencyInfo.denominations[0]
+  } else {
+    const token = tokens[tokenId]
+    currencyCode = token.currencyCode
+    denomination = token.denominations[0]
+  }
+
+  // Fiat Codes
+  const fiatCurrencyCode = wallet.fiatCurrencyCode
+  const isoFiatCurrencyCode = fixFiatCurrencyCode(fiatCurrencyCode)
```

Comment:
I think the first one isn't used? So this could just be `const isoFiatCurrencyCode = fixFiatCurrencyCode(wallet.fiatCurrencyCode)`?

The current way you have it is confusing because the wallet's `fiatCurrencyCode` should already have the "iso:" prefix, but the name implies it does not.